common/tools/dev-tool/src/config/rollup.base.config.ts

```diff
@@ -0,0 +1,118 @@
+  devDependencies: Record<string, string>;
+}
+
+// #region Warning Handler
+
+function ignoreNiseSinonEvalWarnings(warning: RollupWarning): boolean {
+  return warning.code === "EVAL" && (
+    warning.id?.includes("node_modules/nise") ||
+    warning.id?.includes("node_modules/sinon")) === true;
+}
+
+function ignoreChaiCircularDependencyWarnings(warning: RollupWarning): boolean {
+  return warning.code === "CIRCULAR_DEPENDENCY" &&
+      warning.importer?.includes("node_modules/chai") === true;
+}
+
+const warningInhibitors: Array<(warning: RollupWarning) => boolean> = [
+  ignoreChaiCircularDependencyWarnings,
+  ignoreNiseSinonEvalWarnings
+];
+
+/**
+ * Construct a warning handler for the shared rollup configuration
+ * that ignores certain warnings that are not relevant to testing.
+ */
+function makeOnWarnForTesting(): (warning: RollupWarning, warn: WarningHandler) => void {
+  return (warning, warn) => {
+    // If every inhibitor returns false (i.e. no inhibitors), then show the warning
+    if (warningInhibitors.every((inhib) => !inhib(warning))) {
+      warn(warning);
+    }
+  }
+}
+
+// #endregion
+
+function makeBrowserTestConfig() {
+  const config: RollupOptions = {
+    input: ["dist-esm/test/{,!(node)/**/}*.spec.js"],
+    output: {
+      file: `dist-test/index.browser.js`,
+      format: "umd",
+      sourcemap: true,
+      globals: { "fs-extra": "undefined" }
+    },
+    preserveSymlinks: false,
+    // fs-extra must be marked as external in order to avoid an initialization error
+    external: ["fs-extra"],
+    plugins: [
+      multiEntry({ exports: false }),
+      nodeResolve({
+        mainFields: ["module", "browser"],
+        preferBuiltins: true
+      }),
+      cjs({
+        namedExports: {
+          "chai": ["assert", "use"],
+          "@opentelemetry/api": ["CanonicalCode", "SpanKind", "TraceFlags"]
+        }
+      }),
+      json(),
+      sourcemaps(),
+      nodeGlobals(),
+      nodeBuiltinsPlugin()
+      //viz({ filename: "dist-test/browser-stats.html", sourcemap: true })
+    ],
+    onwarn: makeOnWarnForTesting(),
+    // Disable tree-shaking of test code.  In rollup-plugin-node-resolve@5.0.0, rollup started respecting
+    // the "sideEffects" field in package.json.  Since our package.json sets "sideEffects=false", this also
+    // applies to test code, which causes all tests to be removed by tree-shaking.
+    treeshake: false
+  };
+
+  // (config.external as string[]).push(...Object.keys(pkg.devDependencies));
+
+  return config;
+}
+
+export function makeConfig(pkg: PackageJson)  {
+  const baseConfig = {
+    input: "dist-esm/src/index.js",
+    external: [...nodeBuiltins, ...Object.keys(pkg.dependencies), ...Object.keys(pkg.devDependencies)],
+    output: { file: "dist/index.js", format: "cjs", sourcemap: true },
+    preserveSymlinks: false,
+    plugins: [sourcemaps(), nodeResolve(), cjs()]
+  }
+
+  if (IS_PRODUCTION) {
+    baseConfig.plugins.push(terser());
```

Comment:
do we actually use/need this? Since we don't ship browser bundles I'm not sure why we'd want to bother minifying them

Reply:
CC @bterlson